### ISSUE TITLE
[MU4 Issue] [Accessibility] Crash in Style improvements dialog

### ISSUE THREAD

**Issue by shoogle** [musescore/MuseScore#13358]
**Describe the bug**
The _Style improvements_ dialog exhibits the following accessibility problems:

1. No screen reader speech output.
2. Navigation not possible with arrow keys and tab.
3. The OK button doesn't have focus initially.
4. Pressing F6 causes a crash.

**To Reproduce**
1. Save a score in MuseScore 3.6 or earlier.
2. Start a screen reader running.
3. Launch MuseScore 4.
4. Open the 3.6 score you saved previously and attempt to navigate the resulting dialog.

**Expected behavior**
1. There should be screen reader speech output for the controls in the dialog.
2. The tab key should move focus from the checkbox to the buttons. The arrow keys should move focus between the buttons.
3. The OK button should have focus initially so that it can be immediately triggered with Space/Enter/Return.
4. Pressing F6 should do nothing.

**Screenshots**
<img width="592" alt="image" src="https://user-images.githubusercontent.com/11011881/190279637-4ea20f33-e964-4e30-9cfc-befb72fb83a8.png">

**Platform information**
 - OS: macOS, Linux (probably Windows too based on a report from a blind user)

**Comment by MarcSabatella** [comment]
FWIW, the situation seems not so bad on Linux.  Keyboard navigation works, and screen reader reads, and no crash.  There are some awkwardnesses here and there maybe, but overall, it does work.

**Comment by lyrra** [comment]
Can reproduce crash (both windows+linux) pressing F6 in that dialog window.

Backtrace:
678         LOGI() << "nextSec: " << nextSec->name() << ", enabled: " << nextSec->enabled();
(gdb) ba
#0  0x00007ff6aeb50a3a in mu::ui::NavigationController::goToNextSection (this=0x1874b4f8320) at src/framework/ui/internal/navigationcontroller.cpp:678
#1  0x00007ff6aeb4d70a in mu::ui::NavigationController::navigateTo (this=0x1874b4f8320, type=mu::ui::NavigationController::NavigationType::NextSection) at src/framework/ui/internal/navigationcontroller.cpp:391
#2  0x00007ff6aeb4baf2 in operator() (__closure=0x1874c5d7ff0) at src/framework/ui/internal/navigationcontroller.cpp:276

**Comment by wizofaus** [comment]
Can't reproduce F6 crash using latest Nightly on Windows.  Pressing "Esc" does close the dialog for me - that seems expected! - but only when it first shows, not after any other keypress (e.g. F6). Thankfully clicking the top-right X always works, meaning you can in fact realise you're trying to open the wrong file in the wrong app and choose to not proceed.
More seriously though it seems enter/space or any other key to indicate "OK" and open anyway doesn't work.

**Comment by shoogle** [comment]
I'm not sure there's really much value in being able to dismiss the dialog without opening the score, but you're right that the issue with the OK button being inaccessible is more important.

**Comment by wizofaus** [comment]
I'd disagree, because I don't trust that if I open a 3.6 score in v4.0 by mistake, v4.0 might auto-save it and break it for 3.6.

**Comment by cbjeukendrup** [comment]
@wizofaus I don't think that's a real problem: 4.0 opens older scores as if they were imported, so pressing Ctrl+S brings up the Save As… dialog rather than overwriting the 3.6 file. Also, autosave writes to a special .autosave file, and will _never_ overwrite the original file.

(At least, this is how it is designed to work, and as far as I know it works indeed like this.)

**Comment by wizofaus** [comment]
What it's designed to do and what I trust it to do aren't necessarily the same thing! Anyway that was just one example - if I wasn't expecting yo see that dialog because I thought I was opening a v4.0 file in v4 (or even thought I was opening a v3 file in v3) then did so, my default reaction would be to cancel.  It might be some huge file that locks up v4 for a full minute while loading...

**Comment by DmitryArefiev** [comment]
@shoogle I wote to leave Esc how it works now (closing the dialog). We should give user two choices: **open score** and **don't open score** 

If after opening the dialog, focus moves to OK button, then user should just press Enter without any additional navigation actions (@Eism says it can be implemented)

So we will have two scenarios after dialog opens: one Enter to open score or one Esc to don't open.

**Comment by shoogle** [comment]
@DmitryArefiev, sounds good to me! I've updated the description.

**Comment by DmitryArefiev** [comment]
> @DmitryArefiev, sounds good to me! I've updated the description.

@shoogle Oh thanks!

**Comment by DmitryArefiev** [comment]
Fixed in #13416
